libcxx/test/support/test_iterators.h

```diff
@@ -1664,10 +1664,17 @@ using forward_iterator_list = concatenate_t<bidirectional_iterator_list<Ptr>, ty
 template <class Ptr>
 using cpp17_input_iterator_list = concatenate_t<forward_iterator_list<Ptr>, type_list<cpp17_input_iterator<Ptr> > >;
 
+template <class Ptr>
+using cpp17_output_iterator_list = concatenate_t<forward_iterator_list<Ptr>, type_list<cpp17_output_iterator<Ptr> > >;
```

Comment:
Because of the issue we discovered in another PR earlier (about `forward_iterator` not being a valid `output_iterator`), I would avoid creating this typedef, which invites incorrect usage. Instead, I'd just use `types::for_each(types::type_list<cpp17_output_iterator concatenated with forward_iterator_list>(), TestImpl<InIter>());` in `alg.rotate/rotate_copy.pass.cpp`.